lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -174,7 +174,7 @@ def log_metadata_inlining
   # Initially restricted to files sourced from codedir via puppet:/// uri.
   def inline_metadata(catalog, checksum_type)
     environment_path = Pathname.new File.join(Puppet[:environmentpath], catalog.environment, "")
-    environment_path = Puppet::Environments::Directories.real_path(environment_path)
+    environment_path = Pathname.new File.join(Puppet::Environments::Directories.real_path(environment_path), "/")
```

Comment:
It looks like the trailing slash behavior was originally added in db97e88814f8889efd9cf6c6e159392e3c6b5929 so that we could do

https://github.com/puppetlabs/puppet/blob/db97e88814f8889efd9cf6c6e159392e3c6b5929/lib/puppet/indirector/catalog/compiler.rb#L135

Thoughts about removing the magic behavior altogether? I think we'd just need to update the regexp:

https://github.com/puppetlabs/puppet/blob/199e0305c3798e0ef26294127e3a0b4b80f943f6/lib/puppet/indirector/catalog/compiler.rb#L157

To expect the `/` path separator between `environment_path` and `[^\/]+`:

```ruby
metadata.full_path =~ /#{environment_path}\/[^\/]+\/[^\/]+\/#{location}\/.+/
```

The `get_content_uri` method already works as expected if `environment_path` ends with `/` or not:

```ruby
irb(main):022:0> environment_path
=> "/home/josh/.puppetlabs/etc/code/environments/production/"
irb(main):023:0> fullpath
=> "/home/josh/.puppetlabs/etc/code/environments/production/modules/a/files/b.txt"
irb(main):024:0> Pathname.new(fullpath).relative_path_from(environment_path).to_s
=> "modules/a/files/b.txt"
irb(main):025:0> Pathname.new(fullpath).relative_path_from(environment_path.chomp('/')).to_s
=> "modules/a/files/b.txt"
```

Reply:
Yeah, I assumed that we might need to keep that `/` around for better backwards compatibility, but it's definitely a bit cleaner to do away with it entirely and move that trailing slash to the regex. I'll force push the necessary changes to eliminate the trailing slash altogether.